Source/BSN.Commons/Infrastructure/Kafka/KafkaConsumerFactory.cs

```diff
@@ -9,30 +11,49 @@ public class KafkaConsumerFactory<T> : IKafkaConsumerFactory<T>
         /// <param name="options">Default Options for KafkaConsumers</param>
         public KafkaConsumerFactory(IKafkaConsumerOptions options)
         {
-            _defaultConsumerConfig = new ConsumerConfig
+            _defaultConsumerOptions = options;
+            _consumers = new Dictionary<string, IConsumer<Null, T>>();
+        }
+
+        /// <inheritdoc />
+        public IKafkaConsumer<T> Create(string topic, string groupId)
+        {
+            var consumerKey = topic + ":" + groupId;
+
+            if (_consumers.ContainsKey(consumerKey))
+            {
+                return new KafkaConsumer<T>(_consumers[consumerKey]);
+            }
+
+            var config = new ConsumerConfig() 
             {
-                BootstrapServers = options.BootstrapServers,
-                AutoOffsetReset = AutoOffsetReset.Earliest
+                BootstrapServers = _defaultConsumerOptions.BootstrapServers,
+                AutoOffsetReset = AutoOffsetReset.Earliest,
             };
-            
+
+
             // Here we did this because the ReceiveMessageMaxBytes in ProducerConfig type
             // is int and can not accept high values that we expect
-            _defaultConsumerConfig.Set("receive.message.max.bytes", options.ReceiveMessageMaxBytes);
-        }
+            config.Set("receive.message.max.bytes", _defaultConsumerOptions.ReceiveMessageMaxBytes);
 
+            var consumer = new ConsumerBuilder<Null, T>(config).Build();
```

Comment:
describe the meaning of `Null` here with comments

Reply:
done